diff --git a/src/components/app/index.jsx b/src/components/app/index.jsx
index dd5756c..1f44968 100644
--- a/src/components/app/index.jsx
+++ b/src/components/app/index.jsx
@@ -41,12 +41,9 @@ export default function App() {
       <TilePalette
         position={position}
         tileset={tileset}
+        setTileset={setTileset}
         activeTile={activeTile}
         setActiveTile={setActiveTile}
-        size={{
-          width: 640,
-          height: 288,
-        }}
       />
 
       <Map
diff --git a/src/components/tile-palette/index.jsx b/src/components/tile-palette/index.jsx
index a811312..e08c720 100644
--- a/src/components/tile-palette/index.jsx
+++ b/src/components/tile-palette/index.jsx
@@ -1,13 +1,26 @@
 import React from "react";
+import Dropdown from "react-dropdown";
+import "react-dropdown/style.css";
 
 export default function TilePalette({
   tileset,
+  setTileset,
   position,
-  size,
   activeTile,
   setActiveTile,
 }) {
-  const { width, height } = size;
+  const tilesetData = require("../../data/tilesets.json");
+  const tilesets = Object.keys(tilesetData).map((set) => ({
+    type: "group",
+    name: set.split("-").join(" "),
+    items: tilesetData[set].variants.map((variant) => ({
+      value: `${set}/${variant}`,
+      label: variant,
+    })),
+  }));
+
+  const [tilesetGroup, tilesetVariant] = tileset.split("/");
+  const { width, height } = tilesetData[tilesetGroup].size;
   const tiles = [];
   let id = 0;
 
@@ -31,14 +44,23 @@ export default function TilePalette({
         backgroundColor: "white",
       }}
     >
-      <img id="handle" src="/img/drag-handle.png" alt="" />
-      <div
-        style={{
-          background: `url(/sprites/${tileset}.png) -${activeTile.x}px -${activeTile.y}px no-repeat`,
-          width: 32,
-          height: 32,
-        }}
-      />
+      <div style={{ display: "flex" }}>
+        <img id="handle" src="/img/drag-handle.png" alt="" />
+        <div
+          style={{
+            background: `url(/sprites/${tileset}.png) -${activeTile.x}px -${activeTile.y}px no-repeat`,
+            width: 32,
+            height: 32,
+          }}
+        />
+        <div style={{ width: 200 }}>
+          <Dropdown
+            options={tilesets}
+            onChange={(tileset) => setTileset(tileset.value)}
+            value={tileset}
+          />
+        </div>
+      </div>
       {tiles.map((row, y) => (
         <div style={{ display: "flex" }}>
           {row.map((tile, x) => (
diff --git a/src/data/tilesets.json b/src/data/tilesets.json
new file mode 100644
index 0000000..14c0c8b
--- /dev/null
+++ b/src/data/tilesets.json
@@ -0,0 +1,9 @@
+{
+  "rpg-nature-tileset": {
+    "variants": ["spring", "fall", "winter"],
+    "size": {
+      "height": 288,
+      "width": 640
+    }
+  }
+}
